lib/elixir/lib/task.ex

```diff
@@ -304,7 +307,14 @@ defmodule Task do
   again. To await the task's reply multiple times use `yield/2` instead.
   """
   @spec await(t, timeout) :: term | no_return
-  def await(%Task{ref: ref}=task, timeout \\ 5000) do
+  def await(task, timeout \\ 5000)
+
+  # TODO: Remove nil check in Elixir 1.3
+  def await(%Task{owner: owner}=task, _) when owner != nil and owner != self() do
+    raise_invalid_owner_error(task)
```

Comment:
Thats a really good point @jamesfish. I will try to keep an eye for glitches like this one. Thank u